### PR TITLE
Update OTLP to 0.6.0 and use the new Span Status code

### PR DESCRIPTION
**Pull request by tigrannajaryan**
Contributes to https://github.com/open-telemetry/opentelemetry-collector/issues/1841

- Implemented the backward compatibility logic for sending and receiving
  side for DeprecatedCode and Code fields according to OTLP spec.

- Had to modify pdata generator to allow generating fields without setters.
  This is needed because the Status code field needs a custom setter logic
  that implements the DeprecatedCode vs Code handling.

- Added tests that verify OTLP receiver behavior for Status Code/DeprecatedCode.

- Added tests that verify sending/setting behavior for Status Code/DeprecatedCode
  for any component that uses pdata.Traces Status.SetCode().

- Modified all non-OTLP trace receivers and exporters (Jaeger, Zipkin, OpenCensus) to use
  the new Status code.

- Modified correctness tests to use the new status Code instead of DeprecatedCode.

### CODE REVIEW

File: consumer/pdata/trace.go [hunk 1]
```diff
@@ -130,29 +130,69 @@ const (
 	SpanKindCONSUMER    = SpanKind(otlptrace.Span_SPAN_KIND_CONSUMER)
 )
 
+// DeprecatedStatusCode is the deprecated status code used previously.
+// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#set-status
+type DeprecatedStatusCode otlptrace.Status_DeprecatedStatusCode
```
Comment (bogdandrutu) [inline consumer/pdata/trace.go:135]: Consider to also mark the field deprecated.

Comment (tigrannajaryan) [reply consumer/pdata/trace.go:136]: Done.

File: consumer/pdata/trace.go [hunk 2]
```diff
@@ -130,29 +130,70 @@ const (
...
+	//   if code==STATUS_CODE_OK then `deprecated_code` MUST be
+	//   set to DEPRECATED_STATUS_CODE_OK.
+	//
+	//   if code==STATUS_CODE_ERROR then `deprecated_code` MUST be
+	//   set to DEPRECATED_STATUS_CODE_UNKNOWN_ERROR.
+	switch v {
+	case StatusCodeUnset:
+		fallthrough
+	case StatusCodeOk:
+		ms.SetDeprecatedCode(DeprecatedStatusCodeOk)
```
Comment (bogdandrutu) [inline consumer/pdata/trace.go:195]: Just merge these 2 cases instead of using fallthrough?

`case StatusCodeUnset, StatusCodeOk:`

**Review by bogdandrutu — APPROVED** [review]

File: consumer/pdata/trace.go (hunk 2 above)
Comment (tigrannajaryan) [reply consumer/pdata/trace.go]: Fixed.

**Comment by neemiasjnr** [comment]
This may have introduced a bug on opentelemetry-go as linked above, could someone take a look? Thank you!

**Pull request merged**